docs/doxygen/pages.txt

```diff
@@ -51,7 +51,120 @@ The configurations for memory estimation are defined <a href="https://github.com
+these low-level functions can be used directly:
+
+- @ref mqtt_getconnectpacketsize_function <br>
+- @ref mqtt_serializeconnect_function <br>
+- @ref mqtt_getsubscribepacketsize_function <br>
+- @ref mqtt_serializesubscribe_function <br>
+- @ref mqtt_getunsubscribepacketsize_function <br>
+- @ref mqtt_serializeunsubscribe_function <br>
+- @ref mqtt_getpublishpacketsize_function <br>
+- @ref mqtt_serializepublish_function <br>
+- @ref mqtt_serializepublishheader_function <br>
+- @ref mqtt_serializeack_function <br>
+- @ref mqtt_getdisconnectpacketsize_function <br>
+- @ref mqtt_serializedisconnect_function <br>
+- @ref mqtt_getpingreqpacketsize_function <br>
+- @ref mqtt_serializepingreq_function <br>
+- @ref mqtt_deserializepublish_function <br>
+- @ref mqtt_deserializeack_function <br>
+- @ref mqtt_getincomingpackettypeandlength_function <br>
+
+@section mqtt_sessions Sessions and State
+
+The MQTT 3.1.1 protocol allows for a client and server to maintain persistent sessions, which
+can be resumed after a reconnect. The elements of a session stored by this client library consist
+of the states of incomplete publishes with Quality of Service levels of 1 (at least once), or 2 (exactly once).
+These states are stored in the arrays @ref MQTTContext_t.outgoingPublishRecords and @ref MQTTContext_t.incomingPublishRecords;
+the length of each of these arrays can be configured with @ref MQTT_STATE_ARRAY_MAX_COUNT.
+This library does not store any subscription information, nor any information for QoS 0 publishes.
+
+When resuming a persistent session, the client library will resend PUBRELs for all PUBRECs that had been received
+for incomplete outgoing QoS 2 publishes. If the broker does not resume the session, then all state information
+in the client will be reset.
+
+@note The library stores only the <i>state</i> of incomplete publishes and not the publishes themselves.
+If a persistent session is resumed, then @ref mqtt_publishtoresend_function should be called to obtain the
+packet identifiers of incomplete publishes, followed by a call to @ref mqtt_publish_function to resend the
+unacknowledged publish.
+
+@section mqtt_receivepackets Packet Reception
+
+MQTT Packets are received from the network with calls to @ref mqtt_processloop_function or @ref mqtt_receiveloop_function. The functions are mostly identical,
+with the exception of keep-alive; the former sends ping requests and processes ping responses to ensure the keep-alive interval does not expire, while the latter does not.
+Since calls to @ref mqtt_publish_function, @ref mqtt_subscribe_function, and @ref mqtt_unsubscribe_function only send packets and do not
+wait for acknowledgments, a call to @ref mqtt_processloop_function or @ref mqtt_receiveloop_function must follow in order to receive any expected acknowledgment.
+The exception is @ref mqtt_connect_function; since a connection cannot be considered established until the server accepts a CONNECT packet with a CONNACK,
```

Comment:
```suggestion
The exception is @ref mqtt_connect_function; since a MQTT session cannot be considered established until the server acknowledges a CONNECT packet with a CONNACK,
```